internal/cmd/agent.go

```diff
@@ -0,0 +1,139 @@
+package cmd
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"sync"
+	"time"
+
+	"github.com/shirou/gopsutil/v3/process"
+	"github.com/spf13/cobra"
+
+	"github.com/infrahq/infra/internal/logging"
+	"github.com/infrahq/infra/internal/repeat"
+)
+
+// AgentConfig stores details about the agent in config separate from the CLI config file
+// this config file must be separate to avoid concurrent writes with the CLI config
+type AgentConfig struct {
+	ProccessID int `json:"pid"` // used to manage the agent lifecycle
```

Comment:
Does this need to be structured? It's usually enough to write the pid to a file as plaintext, e.g.

```
$ cat agent.pid
12345
```

Reply:
Made this change in [c92fe61](https://github.com/infrahq/infra/pull/2013/commits/c92fe6171674757cade345501c89e3a432c2ba53)